mono/metadata/threads.c

```diff
@@ -1590,7 +1589,7 @@ mono_thread_get_name_utf8 (MonoThread *thread)
 
 	LOCK_THREAD (internal);
 
-	char *tname = g_utf16_to_utf8 (internal->name, internal->name_len, NULL, NULL, NULL);
+	char *tname = (char*)g_memdup (internal->name_utf8, internal->name_len + 1);
```

Comment:
You can use `g_strdup`

Reply:
Probably faster to pass length than recompute it. Strlen is gross.

Reply:
There is another pattern to consider to avoid the implied hypocrisy -- don't store the nul in the internal thread. Allocate one plus length & memcpy length & store nul as needed by others, but not in the internal thread.